tests/testthat/test-sprintf_linter.R

```diff
@@ -91,3 +91,18 @@ test_that("edge cases are detected correctly", {
     linter
   )
 })
+
+test_that("piping into sprintf works", {
+  linter <- sprintf_linter()
+
+  expect_lint("x %>% sprintf(fmt = '%s')", NULL, linter)
+  # no fmt= specified -> this is just 'sprintf("%s", "%s%s")', which won't lint
+  expect_lint('"%s" %>% sprintf("%s%s")', NULL, linter)
+  expect_lint("x %>% sprintf(fmt = '%s%s')", "too few arguments", linter)
+
+  skip_if_not_r_version("4.1.0")
+  # Cannot evaluate statically --> skip
+  expect_lint('x |> sprintf("a")', NULL, linter)
```

Comment:
This one is so important to test we should run it with `%>%` to avoid accidental skips IMO.
I'd also test all the basic cases with the native pipe as well just in case the XML structure differs at any time in the future.

Reply:
parameterized the test